src/servers/src/tls.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2022 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::File;
+use std::io::{BufReader, ErrorKind};
+
+use rustls::{Certificate, PrivateKey, ServerConfig};
+use rustls_pemfile::{certs, pkcs8_private_keys};
+use serde::{Deserialize, Serialize};
+
+// https://www.postgresql.org/docs/current/libpq-ssl.html
+#[derive(Debug, Default, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum TlsMode {
+    #[default]
+    Disable,
+    Prefer,
+    Require,
+    VerifyCa,   // TODO:
+    VerifyFull, // TODO:
+}
+
+#[derive(Debug, Default, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct TlsOption {
+    pub mode: TlsMode,
+    #[serde(default)]
+    pub cert_path: String,
+    #[serde(default)]
+    pub key_path: String,
+}
+
+impl TlsOption {
+    pub fn setup(&self) -> Result<Option<ServerConfig>, std::io::Error> {
+        if let TlsMode::Disable = self.mode {
+            return Ok(None);
+        }
+        let cert = certs(&mut BufReader::new(File::open(&self.cert_path)?))
+            .map_err(|_| std::io::Error::new(ErrorKind::InvalidInput, "invalid cert"))
+            .map(|mut certs| certs.drain(..).map(Certificate).collect())?;
+        let key = pkcs8_private_keys(&mut BufReader::new(File::open(&self.key_path)?))
```

Comment:
So do we require the private key should be in pkcs8 encrypted?

Reply:
`rustls_pemfile` support pkcs1 as well, we can extend more in the future.

Reply:
Got it, can you add a TODO comment here ? For further development.